### PR TITLE
stats: properly extract tag for connection limit filter

### PR DESCRIPTION
**Pull request by youngchul**
Commit Message: stats: properly extract tag for connection limit filter
Additional Description: working towards #21595 
Risk Level: low
Testing: added unit tests to confirm extraction works properly
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

File: changelogs/current.yaml [hunk 1]
```diff
@@ -22,6 +22,11 @@ behavior_changes:
     Fixed tag extraction so that :ref:`stat_prefix <envoy_v3_api_msg_extensions.filters.udp.dns_filter.v3.DnsFilterConfig>`
     is properly extracted. This changes the Prometheus name from
     dns_filter_myprefix_local_a_record_answers{} to dns_filter_local_a_record_answers{envoy.dns_filter_prefix="myprefix"}.
+- area: stats connection_limit
+  change: |
+    Fixed tag extraction so that :ref:`stat_prefix <envoy_v3_api_msg_extensions.filters.network.connection_limit.v3.ConnectionLimit>`
+    is properly extracted. This changes the Prometheus name from envoy_connection_limit_myprefix_limited_connections{} to
+    envoy_connection_limit_limited_connections{envoy_connection_limit_prefix="myprefix"}.
```
Comment (jmarantz) [inline changelogs/current.yaml:29]: Did you mean:
```
envoy_connection_limit.myprefix_limited_connections{} to
envoy_connection_limit.limited_connections{envoy_connection_limit_prefix="myprefix"}.
```
Based on your test it looks like it to me.

**Review by jmarantz — COMMENTED** [review]
lgtm generally.

CI looks broken. Not sure if it's a flake, and I think you have a release-notes typo.

/wait

**Comment by jmarantz** [comment]
Also now that we've started the review, please don't force-push again.

See https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md:
> Once your PR is under review, please do not rebase it. If you rebase, you will need to force push to github, and github's user interface will force your reviewer to review the PR from scratch rather than simply look at your latest changes. It's much easier to review new commits and/or merges. We squash rebase the final merged commit so the number of commits you have in the PR don't matter. Again once your PR is assigned a reviewer, unless you need to fix DCO please do not force push. If you need to pull recent changes you can run
```
branch=$(git status|head -1|cut -f3 -d\ )
git checkout main
git pull
git checkout "$branch"
git merge main
```

File: changelogs/current.yaml (hunk 1 above)
Comment (phlax) [inline changelogs/current.yaml:29]: these should be marked as literals - ie

> ``envoy_connection_limit.myprefix_limited_connections{}`` to ``envoy_connection_limit.limited_connections{envoy_connection_limit_prefix="myprefix"}``.

Comment (youngchul) [reply changelogs/current.yaml:29]: According to [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels), metric names are not allowed to contain dots.  So, `PrometheusStatsFormatter::metricName` replaces dots with underscores.
https://github.com/envoyproxy/envoy/blob/21323d483ed163155fb8ceba0935c33c6dae7a2f/source/server/admin/prometheus_stats.cc#L311
https://github.com/envoyproxy/envoy/blob/21323d483ed163155fb8ceba0935c33c6dae7a2f/source/server/admin/prometheus_stats.cc#L24-L29

I believe the following is accurate.
```
``envoy_connection_limit_myprefix_limited_connections{}`` to
``envoy_connection_limit_limited_connections{envoy_connection_limit_prefix="myprefix"}``.
```

Comment (youngchul) [reply changelogs/current.yaml:29]: Thanks. I will add backticks for metric names.

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!
Assigning Greg for senior-maintainer approval.
/assign @ggreenway 

@jmarantz @phlax any other comments?

**Review by phlax — COMMENTED** [review]
lgtm from docs pov, thanks @youngchul

**Review by ggreenway — APPROVED** [review]

**Pull request merged**